lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -1002,7 +1002,7 @@ def modify_connection_ethernet(self, conn_type='ethernet'):
             'autoconnect': self.bool_to_string(self.autoconnect),
             'ipv4.dns-search': self.dns4_search,
             'ipv6.dns-search': self.dns6_search,
-            '802-3-ethernet.mtu': self.mtu,
+            '802-3-ethernet.mtu': str(self.mtu),
```

Comment:
How about adding `to_text(value)` in next loop where we are iterating on options ? 

```python
        for key, value in options.items():
            if value is not None:
                if key == '802-3-ethernet.mtu' and conn_type != 'ethernet':
                    continue
                cmd.extend([key, to_text(value)])
```

Reply:
> How about adding `to_text(value)` in next loop where we are iterating on options ?
> 
> ```python
>         for key, value in options.items():
>             if value is not None:
>                 if key == '802-3-ethernet.mtu' and conn_type != 'ethernet':
>                     continue
>                 cmd.extend([key, to_text(value)])
> ```

Importing the `to_text` function
```python
from ansible.module_utils._text import to_text
```
and replacing
```python
        for key, value in options.items():
            if value is not None:
                cmd.extend([key, value])
```
with
```python
        for key, value in options.items():
            if value is not None:
                cmd.extend([key, to_text(value)])
```
did solve issue https://github.com/ansible/ansible/issues/59095 for me. I however don't know what the consequences are.

Reply:
@Akasurde I've made a mistake during previous rebase and didn't notice that change of `str` to `to_text` was lost. I've fixed that right now.

As mentioned in previous comment - the converion is done for every parameter explicitly on purpose.
I've decided to do this only on the values which are expected to be non-string and will correctly convert using to_text (boolean, for example, won't convert correctly).
For this reason (correctness) I think this should be done for each value, not for all of them.

However, if it is preffered, I will change the implementation to one that will do this for all arguments (in the loop). Just let me know.

@polatsinan Does the issue persist when applying current changes in the PR?

Reply:
@marcin-sucharaski check against current development, I've merged stringification at run_command so you might not need to do this anymore.

Reply:
@bcoca I have cloned the devel branche and the issue still exists:
```
ansible 2.9.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/spolat/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/spolat/python/ansible/local/lib/python2.7/site-packages/ansible
  executable location = /home/spolat/python/ansible/bin/ansible
  python version = 2.7.15+ (default, Nov 27 2018, 23:36:35) [GCC 7.3.0]
```

```
TypeError: argument of type 'int' is not iterable
```

Reply:
that is not same issue as original above